.travis.yml

```diff
@@ -1,61 +1,137 @@
+# Tell Travis to use new container-based infrastructure
 sudo: false
 
+# Declare project language.
+# @link http://about.travis-ci.org/docs/user/languages/php/
 language: php
 
-notifications:
-  email:
-    on_success: never
-    on_failure: change
-
+# Specify when Travis should build.
 branches:
   only:
     - master
 
+# Git clone depth.
+git:
+  depth: 1
+
 matrix:
   include:
-    - php: 5.3
-      env: WP_VERSION=latest
-      dist: precise
     - php: 5.6
-      env: WP_VERSION=latest
+      env: WP_VERSION=latest PHP_LINT=1
     - php: 7.0
-      env: WP_VERSION=latest
+      env: WP_VERSION=latest PHP_LINT=1
     - php: 7.1
-      env: WP_VERSION=latest
+      env: WP_VERSION=latest PHP_LINT=1
     - php: 7.2
-      env: WP_VERSION=latest
-    - php: 5.6
-      env: WP_VERSION=nightly
-    # - php: 5.6
-    #   env: WP_TRAVISCI=phpcs
+      env: WP_VERSION=latest PHP_LINT=1 WP_PHPCS=1
   fast_finish: true
 
+# Use this to prepare the system to install prerequisites or dependencies.
+# e.g. sudo apt-get update.
+# Failures in this section will result in build status 'errored'.
+before_install:
+
+# Use this to prepare your build for testing.
+# e.g. copy database configurations, environment variables, etc.
+# Failures in this section will result in build status 'errored'.
 before_script:
+  # Turn off Xdebug. See https://core.trac.wordpress.org/changeset/40138.
+  - phpenv config-rm xdebug.ini || echo "Xdebug not available"
+
   - export PATH="$HOME/.composer/vendor/bin:$PATH"
+
+  # Couple the PHPUnit version to the PHP version.
   - |
-      if [[ ${TRAVIS_PHP_VERSION:0:2} == "7." ]]; then
-        composer global require "phpunit/phpunit=5.7.*"
-      else
+    case "$TRAVIS_PHP_VERSION" in
+      7.*)
+        echo "Using PHPUnit 6.1"
+        composer global require "phpunit/phpunit=6.1.*"
+        ;;
+      *)
+        echo "Using PHPUnit 4.8"
         composer global require "phpunit/phpunit=4.8.*"
-      fi
-  - composer install
-  - |
-    if [[ ! -z "$WP_VERSION" ]] ; then
-      bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION
-    fi
```

Comment:
personally, I'm a fan of the shell script approach. why the change?

Reply:
Better alignment with other Travis configs that we use.

Reply:
Most of our themes take this approach, but I _think_ most if not all plugins use the shell script. Themes probably should use it too, though.

Reply:
Good call—I can restore this in my next PR.